testkit/src/builder.rs

```diff
@@ -117,34 +114,20 @@ pub struct TestKitBuilder {
     our_validator_id: Option<ValidatorId>,
     test_network: Option<TestNetwork>,
     logger: bool,
-    rust_services: Vec<Box<dyn ServiceFactory>>,
-    runtimes: Vec<(u32, Box<dyn Runtime>)>,
+    rust_runtime: RustRuntime,
+    additional_runtimes: HashMap<u32, Box<dyn Runtime>>,
     instances: Vec<InstanceConfig>,
 }
 
 impl TestKitBuilder {
     /// Creates testkit for the validator node.
     pub fn validator() -> Self {
-        TestKitBuilder {
-            test_network: None,
-            our_validator_id: Some(ValidatorId(0)),
-            logger: false,
-            rust_services: vec![],
-            runtimes: vec![],
-            instances: vec![],
-        }
+        Self::new(true)
```

Comment:
Nit: `Self::new(true)` does not really explain what exactly is going on. Maybe add an enum `TestkitRole` with variants `Validator` and `Auditor`?

Reply:
Well, good point, thanks. But it's rather an internal method that eliminates code duplication for the `validator()` and `auditor()` methods. And now I think why do we need that boolean flag and why not just pass the Option<ValidatorId> instead? Going to implement it that way.

Reply:
Yeah, makes sense.